include/aws/s3/private/s3_auto_ranged_put.h

```diff
@@ -69,11 +75,18 @@ struct aws_s3_auto_ranged_put {
         struct aws_s3_paginated_operation *list_parts_operation;
         struct aws_string *list_parts_continuation_token;
 
+        /* Note: total num parts is known only if content-length is known,
+        otherwise it will be 0. */
         uint32_t total_num_parts;
         uint32_t num_parts_sent;
+        uint32_t num_parts_prepared;
```

Comment:
"prepared" is a bit ambiguous. We increment this in the middle of the "prepare request" pipeline. There's still more work before we're done preparing, like async signing. I'd expect something like this to be incremented after the "prepare request" pipeline is finished, right before the callback fires

Also, an aws_s3_request can be prepared multiple times, in the case of retry (we skip reading if its not the first time we've prepared it)

maybe?
```suggestion
        /* Number of "sent" parts we've finished reading the body for
         * (does not include skipped parts in the case of pause/resume) */
        uint32_t num_parts_read;
```